features/TESTS/filesystem/flashsim_block_device/main.cpp

```diff
@@ -0,0 +1,135 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "mbed.h"
+#include "greentea-client/test_env.h"
+#include "unity.h"
+#include "utest.h"
+
+#include "FlashSimBlockDevice.h"
+#include "HeapBlockDevice.h"
+#include <stdlib.h>
+
+using namespace utest::v1;
+
+static const bd_size_t read_size = 1;
+static const bd_size_t prog_size = 8;
+static const bd_size_t erase_size = 512;
+static const bd_size_t num_blocks = 4;
+static const bd_size_t test_buf_size = 64;
+static const uint8_t   blank = 0xFF;
+
+// Simple test for all APIs
+void functionality_test() {
```

Comment:
`{`  new line in this test file for function body

Reply:
Done.